cluster_configs/barnacle/barnacle_status.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+
+import subprocess
+import sys
+import time
+import re
+from io import StringIO
+
+def parse_qstat(lines):
+    status = {}
+
+    for line in lines:
+        if line.startswith(' '):
+            entry = line.rstrip()
+            p = re.compile('^ +(.+?) = (.+)')
+            m = p.match(entry.rstrip())
+            status[m.groups()[0]] = m.groups()[1]
+        elif line.startswith('\t'):
+            entry += line.strip()
+            p = re.compile('^ +(.+?) = (.+)')
+            m = p.match(entry.rstrip())
+            status[m.groups()[0]] = m.groups()[1]
+
+    return(status)
+
+def print_status(status):
```

Comment:
I kind-of feel that a function should better return a value rather than interact with the user. Instead, `main()` could print what is return by this function.